rotkehlchen/balances/historical.py

```diff
@@ -120,6 +121,67 @@ def get_assets_amounts(
 
         return amounts, last_event_id
 
+    def get_historical_netvalue(
+            self,
+            from_ts: Timestamp,
+            to_ts: Timestamp,
+    ) -> tuple[dict[Timestamp, FVal], list[tuple[str, Timestamp]]]:
+        """Calculates historical net worth per day within the given time range.
+
+        Uses asset balances per day combined with historical prices to calculate
+        the total net worth in the user's main currency for each day.
+
+        May raise:
+            - NotFoundError if no events exist in the specified time period.
+
+        Returns:
+            - A mapping of timestamps to net worth in main currency for each day
+            - A list of (asset id, timestamp) tuples where price data was missing
+        """
+        events, main_currency = self._get_events_and_currency(from_ts=from_ts, to_ts=to_ts)
+        if len(events) == 0:
+            raise NotFoundError(f'No historical data found within {from_ts=} and {to_ts=}.')
+
+        current_balances: dict[Asset, FVal] = defaultdict(FVal)
+        daily_balances: dict[Timestamp, dict[Asset, FVal]] = defaultdict(lambda: defaultdict(FVal))
```

Comment:
this doesn't need to be a defaultdict as you use it right now unless I'm missing something